File: sample/sample.js
//
// KONKER SAMPLE APPLICATION TO LIST ALL DEVICES ASSOCIATED WITH AN ACCOUNT 
// 2019
//
// MIT License
// 

const konker = require('../src/nodejs/KonkerApi');

const username = 'jobs@konkerlabs.com';
const password = 'gokonkergokonker!';
const appname = 'default';


konker.authenticate(username, password).then((res) => {
  konker.getAllDevices(appname).then((res) => {
    // console.log(res.data);
    if (res.data.code === 200) { 
      // console.log(res.data);
      res.data.result.forEach(device => {

        console.log('device = '+ JSON.stringify(device));

        konker.getIncomingEvents(appname, device.guid).then(res => {
          console.log('DEVICE DATA --> ');
          console.log(res.data);
        })

      });

    }
  }).catch(err => {console.log(err)})
}).catch(err => {console.log(err)})

File: sample/sample.py
'''
 KONKER SAMPLE APPLICATION TO LIST ALL DEVICES ASSOCIATED WITH AN ACCOUNT 
 2019

 MIT License
'''

import sys
sys.path.insert(0, '../src/python')

import konker as api

konker = api.Konker()

username = 'jobs@konkerlabs.com'
password = 'gokonkergokonker!'

(_, token) = konker.login(username=username, password=password)

if token:

  print('LIST OF AVAILABLE DEVICES ------ ')
  for device in konker.getAllDevices(): 
    print(device)

    print('. GETTING DATA FOR DEVICE {} ---- '.format(device['name']))
    data = konker.readData(device['guid'], delta=-1)
    print(data)
    print('...')

  print ('Done.')
